ui-tests/src/test/java/org/openmrs/reference/EndVisitTest.java

```diff
@@ -33,20 +32,20 @@ public void setUp() throws Exception {
     }
 
     @Test
-    @Ignore
     @Category(BuildTests.class)
-    public void EndVisitTest() throws Exception {
+    public void endVisitTest() {
         FindPatientPage findPatientPage = homePage.goToFindPatientRecord();
         findPatientPage.enterPatient(patient.identifier);
+        findPatientPage.waitForPageToLoad();
         ClinicianFacingPatientDashboardPage clinicianFacingPatientDashboardPage = findPatientPage.clickOnFirstPatient();
         PatientVisitsDashboardPage patientVisitsDashboardPage = clinicianFacingPatientDashboardPage.goToRecentVisits();
         patientVisitsDashboardPage = patientVisitsDashboardPage.endVisit();
+        patientVisitsDashboardPage.waitForPageToLoad();
```

Comment:
`Waiting the page to load` ensures that the `endVist()` operation is effective otherwise by the time `assertNull()` applies its logic to finds weather the active visit is `null` unfortunately finds it still active due to delay of the `endVisit()` operation on the visit

Reply:
when would you use `waitForPage()` vs `waitForPageToLoad()`?

Reply:
> when would you use `waitForPage()` vs `waitForPageToLoad()`?

I would use [waitforPage()](https://github.com/openmrs/openmrs-contrib-uitestframework/blob/e3293792449c062b4934f0c29db8cb02f9156613/src/main/java/org/openmrs/uitestframework/page/Page.java#L119) when the page takes less than 500 milliseconds to load since the method implementation extends a default `TimeUnit `
(500 milliseconds) from the selenium API. However, [waitforPageToLoad()](https://github.com/openmrs/openmrs-contrib-uitestframework/blob/e3293792449c062b4934f0c29db8cb02f9156613/src/main/java/org/openmrs/uitestframework/page/Page.java#L165) is customized to a `TimeUnit` of 1000 milliseconds and this gives a room for pages that take longer than 500 milliseconds to load to rather implement `waitforPageToLoad()` over` waitforPage()`.